File: 01_easy/two_sum/two_sum_brute_force.py
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(0, len(nums)-1):
            for j in range(i+1, len(nums)):
                if nums[i] + nums[j] == target:
                    return [i, j]
        return []


if __name__ == "__main__":
    s = Solution()
    result = s.twoSum([2, 3, 11, 15], 14)
    print(result)

File: 01_easy/two_sum/two_sum_brute_force.cpp

#include <iostream>
#include <vector>

std::vector<int> two_sum(std::vector<int> nums, const int target)
{
    for (int i = 0; i < nums.size(); ++i) {
        for (int j = i + 1; j < nums.size(); ++j) {
            if (nums[i] + nums[j] == target) {
                return {i, j};
            }
        }
    }

    return {};
}

int main()
{
    std::vector<int> nums = {1, 2, 3, 4};

    auto idx_vec = two_sum(nums, 7);

    for (auto idx : idx_vec) {
        std::cout << idx << std::endl;
    }    

    return 0;
}